### PR TITLE
Distributors rings status

### LINKED ISSUE #3656: "Ring status" for distributors
**Is your feature request related to a problem? Please describe.**
It's very easy to see ingester and ha ring status, but distributor ring status is only _sort of_ viewable via Consul

**Describe the solution you'd like**
A status page that shows the distributor ring status similar to how we can currently view ingester ring status.

**Describe alternatives you've considered**
Currently I take a look at the distributor key in consul.

**Additional context**
It would just help with troubleshooting if distributors are having trouble creating/joining.

### PR DESCRIPTION
**Pull request by ethervoid**
**What this PR does**:

Now we're able to see the ring status for the distributor when the ingestion-rate-limit-strategy
the setting is set to global

Screenshot from 2021-04-30 18-58-34
Screenshot from 2021-04-30 18-57-59
Screenshot from 2021-04-30 18-57-48

**Which issue(s) this PR fixes**:
Fixes #3656

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by ethervoid** [comment]
@pracucci Finally I've been able to do it :smile: , hopefully, I haven't done something totally weird that is not what we want

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 * [CHANGE] Alertmanager: allowed to configure the experimental receivers firewall on a per-tenant basis. The following CLI flags (and their respective YAML config options) have been changed and moved to the limits config section: #4143
   - `-alertmanager.receivers-firewall.block.cidr-networks` renamed to `-alertmanager.receivers-firewall-block-cidr-networks`
   - `-alertmanager.receivers-firewall.block.private-addresses` renamed to `-alertmanager.receivers-firewall-block-private-addresses`
+* [CHANGE] Distributor: Added ring status section in the admin page #
```
Comment (pracucci) [inline CHANGELOG.md:9]:
```suggestion
* [CHANGE] Distributor: Added ring status section in the admin page. #4151
```

File: pkg/distributor/distributor.go [hunk 2]
```diff
@@ -202,33 +203,39 @@ func New(cfg Config, clientConfig ingester_client.Config, limits *validation.Ove
...
 		ingestionRateStrategy = newInfiniteIngestionRateStrategy()
 	} else if limits.IngestionRateStrategy() == validation.GlobalIngestionRateStrategy {
-		distributorsRing, err = ring.NewLifecycler(cfg.DistributorRing.ToLifecyclerConfig(), nil, "distributor", ring.DistributorRingKey, true, reg)
+		distributorsLifeCycler, err = ring.NewLifecycler(cfg.DistributorRing.ToLifecyclerConfig(), nil, "distributor", ring.DistributorRingKey, true, reg)
 		if err != nil {
 			return nil, err
 		}
 
-		subservices = append(subservices, distributorsRing)
+		distributorsRing, err = ring.New(cfg.DistributorRing.ToRingConfig(), "distributor", ring.DistributorRingKey, prometheus.DefaultRegisterer)
```
Comment (pracucci) [inline pkg/distributor/distributor.go:217]: Please don't use `prometheus.DefaultRegisterer` but always the provided one.

```suggestion
		distributorsRing, err = ring.New(cfg.DistributorRing.ToRingConfig(), "distributor", ring.DistributorRingKey, reg)
```

File: pkg/distributor/distributor.go [hunk 3]
```diff
@@ -202,33 +203,39 @@ func New(cfg Config, clientConfig ingester_client.Config, limits *validation.Ove
...
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to initialize distributors' ring")
```
Comment (pracucci) [inline pkg/distributor/distributor.go:219]: [nit]

```suggestion
			return nil, errors.Wrap(err, "failed to initialize distributors' ring client")
```

File: pkg/distributor/distributor.go [hunk 4]
```diff
@@ -1029,3 +1036,27 @@ func (d *Distributor) AllUserStats(ctx context.Context) ([]UserIDStats, error) {
...
+		var unshardedPage = `
+			<!DOCTYPE html>
+			<html>
+				<head>
+					<meta charset="UTF-8">
+					<title>Cortex Distributor Status</title>
+				</head>
+				<body>
+					<h1>Cortex Distributor Status</h1>
+					<p>Distributor is not running with global limits disabled</p>
```
Comment (pracucci) [inline pkg/distributor/distributor.go:1053]:
```suggestion
					<p>Distributor is not running with global limits enabled.</p>
```

**Review by pracucci — APPROVED** [review]
Good job, LGTM! (modulo a couple of nits)

File: pkg/distributor/distributor.go [hunk 5]
```diff
@@ -1029,3 +1036,27 @@ func (d *Distributor) AllUserStats(ctx context.Context) ([]UserIDStats, error) {
...
+				</body>
+			</html>`
+		w.WriteHeader(http.StatusOK)
+		_, err := w.Write([]byte(unshardedPage))
+		if err != nil {
+			level.Error(d.log).Log("msg", "unable to serve status page", "err", err)
+		}
+	}
```
Comment (pstibrany) [inline pkg/distributor/distributor.go:1061]: Is this worth moving to `WriteHTMLResponse` in `pkg/util/http.go`?

Also there is no need to log the error while writing response. It's likely that client has disconnected, at that point, we cannot do anything about it.

**Review by pstibrany — APPROVED** [review]
LGTM, thanks (let's please fix `prometheus.DefaultRegisterer` that @pracucci pointed out).

File: pkg/distributor/distributor.go (hunk 5 above)
Comment (ethervoid) [reply pkg/distributor/distributor.go]: Yeah it makes sense so I've created that method and now we're using it in rules and distributor :)

File: pkg/distributor/distributor.go (hunk 2 above)
Comment (ethervoid) [reply pkg/distributor/distributor.go]: Oh I see, great catch :)

**Comment by ethervoid** [comment]
All changes tackled :)

**Comment by pstibrany** [comment]
Thanks for addressing our feedback!

**Pull request merged**